### PR TITLE
pallet-collective: Do not vote `aye` with `propose`

### LINKED ISSUE #6558: add RetractVote to Council voting process
To send a treasury proposal to the Council, members call `Propose` in the Council module - this sends the proposal to vote by the Council **and** records the vote of the sender. This causes an issue when the Prime Member sends a proposal, because by signalling their vote councillors complain about the Prime Member signalling a decision in the best case scenario and some rely on soft-voting to pass motions in the worst. 

I'd like to:
1. add `RetractVote` call for the Prime Member to use (and vote aye or nay later when needed), or
2. allow the Council member to send proposals to council for voting without adding their vote in the same call.

### PR DESCRIPTION
**Pull request by emostov**
Closes #6558 

This is a breaking change for the `propose` extrinsic behavior. 

Currently, when executing `propose`, the caller records an `aye` vote for the proposal. This PR removes that logic so no votes are  recorded when `propose` executes.

### CODE REVIEW

File: frame/collective/src/benchmarking.rs [hunk 1]
```diff
@@ -619,6 +626,7 @@ benchmarks_instance! {
 				threshold,
 				Box::new(proposal.clone()),
 				bytes_in_storage,
+				true, // Include aye vote on behalf of proposer.
 			)?;
 			last_hash = T::Hashing::hash_of(&proposal);
 		}
```
Comment (emostov) [inline frame/collective/src/benchmarking.rs:632]: Do we want to try a benchmark case for not including an aye vote?

**Comment by shawntabrizi** [comment]
> Is it worth simply removing the logic to record an aye for the caller and instead always expect them to follow up vote to declare their position?

IMO yes, this should be the behavior, plus it wont break the transaction versioning as this PR does.

**Review by kianenigma — APPROVED** [review]
I think a `retract_vote` would also be nice in general.

**Comment by kianenigma** [comment]
> > Is it worth simply removing the logic to record an aye for the caller and instead always expect them to follow up vote to declare their position?
> 
> IMO yes, this should be the behavior, plus it wont break the transaction versioning as this PR does.

to be very pedantic, since the underlying behaviour of a transaction is changing, that's still a transaction version change.

**Comment by emostov** [comment]
> > > Is it worth simply removing the logic to record an aye for the caller and instead always expect them to follow up vote to declare their position?
> > 
> > 
> > IMO yes, this should be the behavior, plus it wont break the transaction versioning as this PR does.
> 
> to be very pedantic, since the underlying behaviour of a transaction is changing, that's still a transaction version change. I know that we mostly use transaction version for instances where a parameter is added or removed, but that's not exactly true.

Do we have a definition for transaction version somewhere? AFAICT we use to signify a broken call index or broken call signature so wallets can get a heads up. But I have not seen it used for a logic change (although I think we should have some versioning reflect breaking changes to logic.)

**Comment by shawntabrizi** [comment]
> to be very pedantic, since the underlying behaviour of a transaction is changing, that's still a transaction version change. I know that we mostly use transaction version for instances where a parameter is added or removed, but that's not exactly true.

Sorry I disagree here. This is a runtime upgrade and a `spec_version` change. Behavior of transactions and the whole runtime in general can be modified between `spec_versions`. Transaction version points at the compatibility that an encoded transaction will be successfully processed by the node. (maybe there is something a little more specific than that, but this is the rough definition in my head)

**Review by shawntabrizi — APPROVED** [review]
lgtm. need to let benchmarks finish.

cc @rrtti

**Comment by shawntabrizi** [comment]
bot merge

**Review by shawntabrizi — APPROVED** [review]

**Comment by freehere107** [comment]
@emostov Hello, Ask a question, is there any way to know whether the chain has updated this pr, subscan Here, when propose is created, vote aye+1 by default

**Comment by emostov** [comment]
@freehere107  It is not on Polkadot or Kusama yet - it will be though once they update to runtime v9090 or above.

For something like this it should get listed in the release notes. I checked and its in v0.9.9 (see the notes here https://github.com/paritytech/polkadot/releases/tag/v0.9.9); which corresponds to runtime v9090 for polkadot & kusama. To  figure out the current runtime you can query the `RuntimeVersion`, but even easier if going to polkadot.js apps and looking in the upper left hand corner. Both polkadot & kusama are on runtime v9080, so this change has not taken affect

**Comment by freehere107** [comment]
For polkadot, kusama, and subscan, this can be accurate to the spec version. But for other networks, we are not sure when they will merge this pr. I think this break change should be more convenient to add storage like the previous ``accountInfo `` update.

**Comment by shawntabrizi** [comment]
@freehere107 you should be able to query the aye / nay from the storage directly rather than making this assumption.

**Comment by freehere107** [comment]
> @freehere107 you should be able to query the aye / nay from the storage directly rather than making this assumption.

@shawntabrizi  Yes, thank you for reminding me, votes can get from chain storage

**Pull request merged**